utils/onnxmlirrun.py

```diff
@@ -0,0 +1,186 @@
+# "standalone" python script to use onnx-mlir compiler.
+# In future, this file will evolved to be part of onnx-mlir python package. 
+# An example:
+
+"""
+import onnxmlirrun
+import numpy as np
+
+a = np.random.rand(3, 4, 5).astype(np.float32)
+b = np.random.rand(3, 4, 5).astype(np.float32)
+session = onnxmlirrun.InferenceSession("test_add.onnx")
+outputs = session.run(None, {"a": a, "b":b})
+print(outputs)
+"""
+
+
+
+class InferenceSession:
+
+    def __init__(self, model_path, target="cpu", **kwarg):
+        self.target = target
+        if "options" in kwarg :
+          self.options = kwarg["options"]
+        else :
+          self.options = ""
+
+        print("target : ", self.target)
+        print(" options : ", self.options)
+
+        # Initialize status
+
+        self.compiled = False
+        self.loaded = False
+
+        # Initialize parameters
+
+        self.VERBOSE = os.environ.get('VERBOSE', False)
+        self.input_model_path = model_path
+
+        # name for the compiled library in temporary directory
+
+        self.temp_lib_name = 'model'
+        if not os.environ.get('ONNX_MLIR_HOME', None):
+            raise RuntimeError('Environment variable ONNX_MLIR_HOME is not set, please set it to the path to the HOME directory for onnx-mlir. The HOME directory for onnx-mlir refers to the parent folder containing the bin, lib, etc sub-folders in which ONNX-MLIR executables and libraries can be found, typically `onnx-mlir/build/Debug`'
+                               )
+
+        self.ONNX_MLIR_EXENAME = 'onnx-mlir'
+        if sys.platform == 'win32':
+            self.ONNX_MLIR_EXENAME = 'onnx-mlir.exe'
+
+        # Compiler package related parameters.
+        # Should be changed when package is installed
+
+        self.ONNX_MLIR = os.path.join(os.environ['ONNX_MLIR_HOME'],
+                'bin', self.ONNX_MLIR_EXENAME)
+        self.RUNTIME_DIR = os.path.join(os.environ['ONNX_MLIR_HOME'],
+                'lib')
+        sys.path.append(self.RUNTIME_DIR)
+        try:
+            from PyRuntime import OMExecutionSession
+        except ImportError:
+            raise ImportError('Looks like you did not build the PyRuntime target, build it by running `make PyRuntime`.You may need to set ONNX_MLIR_HOME to `onnx-mlir/build/Debug` since `make PyRuntime` outputs to `build/Debug` by default'
+                              )
+
+    def compile(self):
+
+        # Prepare compiler arguments.
+
+        self.temp_dir = tempfile.TemporaryDirectory()
+        command_str = [self.ONNX_MLIR]
+
+        # for onnxruntime, the provider flag will determine the flags
+        # need more work on flags here
+
+        command_str += [self.input_model_path]
+        output_path = os.path.join(self.temp_dir.name,
+                                   self.temp_lib_name)
+        command_str += ['-o', output_path]
+        if self.target == 'zAIU' :
+            command_str += ['--maccel=NNPA']
```

Comment:
FYI target `zAIU` should also trigger `-mcpu=z16`. We also strongly recommend to use `-O3` default.

Reply:
Added